### PR TITLE
fixing marginal custom CI in predictive distributions

### PR DESCRIPTION
**Pull request by FBartos**
the else statement was supposed to be tied to the first if instead of the second - automatically deleting the passed data - fixed by adding curly brackets

### CODE REVIEW

File: JASP-Engine/JASP/R/LSbinomialcommon.R [hunk 1]
```diff
@@ -594,20 +594,21 @@
...
+  if (!is.null(density)){
+    if (!densityDiscrete)
+      density$y <- density$y/nrow(density)    
+  }else
+    density <- data.frame("y" = NULL, "x" = NULL)
+  
+  if (length(spikes) != 0){
+    for(i in 1:length(spikes)){
+      density <- rbind(density[density$x <= spikes[[i]]$x,], spikes[[i]], density[spikes[[i]]$x < density$x,])
     }
```
Comment (Kucharssim) [inline JASP-Engine/JASP/R/LSbinomialcommon.R]:
```suggestion
  for(i in seq_along(spikes)) {
    density <- rbind(density[density$x <= spikes[[i]]$x,], spikes[[i]], density[spikes[[i]]$x < density$x,])
  }
```

File: JASP-Engine/JASP/R/LSbinomialcommon.R [hunk 2]
```diff
@@ -594,20 +594,21 @@
...
-      density <- data.frame("y" = NULL, "x" = NULL)
-    
-    if (length(spikes) != 0){
-      for(i in 1:length(spikes)){
-        density <- rbind(density[density$x <= spikes[[i]]$x,], spikes[[i]], density[spikes[[i]]$x < density$x,])
-      }
+  if (!is.null(density)){
+    if (!densityDiscrete)
+      density$y <- density$y/nrow(density)    
+  }else
```
Comment (Kucharssim) [inline JASP-Engine/JASP/R/LSbinomialcommon.R]: This is still a bit iffy, I believe you it works now but can you simplify the logic a little to remove these nested if statemenets? I won't beat you on this but it's definitely not pretty...

File: JASP-Engine/JASP/R/LSbinomialcommon.R (hunk 1 above)
Comment (Kucharssim) [reply JASP-Engine/JASP/R/LSbinomialcommon.R]: if `length(spikes) == 0`, `seq_along(spikes)` gives you `numeric(0)` and the loop is skipped, no need to nest it like this

**Review by Kucharssim — CHANGES_REQUESTED** [review]

File: JASP-Engine/JASP/R/LSbinomialcommon.R (hunk 1 above)
Comment (FBartos) [reply JASP-Engine/JASP/R/LSbinomialcommon.R]: true, thanks (y)

File: JASP-Engine/JASP/R/LSbinomialcommon.R (hunk 2 above)
Comment (FBartos) [reply JASP-Engine/JASP/R/LSbinomialcommon.R]: yeah, I think that it's a remnant from a time when the densityDiscrete was a part the density object itself. I will change it.

**Review by Kucharssim — APPROVED** [review]
Assuming the tests are ok, it's fine to merge

**Comment by FBartos** [comment]
I merged the commits together, but I can't build/test the release version of JASP, I have some problems with the build...

**Comment by TimKDJ** [comment]
Does this close an issue?

**Comment by FBartos** [comment]
it does not, I fixed it directly without creating an issue

**Pull request merged**